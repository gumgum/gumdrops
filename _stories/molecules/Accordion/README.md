The `<Accordion>` component is a collapsible container for holding related elements. Nest `<AccordionItem>` components inside `<Accordion>` for each drawer. Nest `<AccordionItemContent>` in each `<AccordionItem>` if you would like to display listed information within an open `<AccordionItem>`. Each `<Accordion>` related component accepts a className and otherProps so you are able to further customize the component.

## Accordion Props

The following props may be passed to configure the `<Accordion/>`:

| name           | type      | description                                                                             | Required |
| -------------- | --------- | --------------------------------------------------------------------------------------- | -------- |
| children       | `Node`    | Children should be `<AccordionItem>`                                                    | Yes      |
| context        | `String`  | Indicate the context of the Accordion. One of `dark`, `white`, or `undefined`           |          |
| initialAllOpen | `Boolean` | Indicate if all `<AccordionItem>` children should be opened or closed on initial mount. |          |
| size           | `String`  | Indicate the size of the Accordion. One of `sm` or `undefined`                          |          |

## AccordionItem Props

The following props may be passed to configure the individual `<AccordionItem />` components:

| name     | type      | description                                                                       |
| -------- | --------- | --------------------------------------------------------------------------------- |
| context  | `String`  | Indicate the context of the AccordionItem. One of `dark`, `white`, or `undefined` |
| isOpen   | `Boolean` | Indicate if the item should be opened or closed on initial mount.                 |
| isLocked | `Boolean` | Indicate if the item should be locked open or closed and ignore user interaction  |
| label    | `String`  | The text label for an item                                                        |
| size     | `String`  | Indicate the size of the AccordionItem. One of `sm` or `undefined`                |

## Examples

You can use `<Accordion>` to display any information that you pass in, such as:

```
<Accordion>
    <AccordionItem label="Controls">
        <FormGroup>
            <FormGroupLabel text="Name" />
            <TextInput name="name" />
        </FormGroup>
    </AccordionItem>
</Accordion>
```

You can also use `<Accordion>` to display information in a list form, in the case you want to list items in rows. Example:

```
<Accordion>
    <AccordionItem label="Settings">
        <AccordionItemContent>
            <Column md="6">
                Setting 1
            </Column>
            <Column md=6>
                <Button>Save</Button>
            </Column>
        </AccordionItemContent>
        <AccordionItemContent>
            <Column md="6">
                Setting 2
            </Column>
            <Column md=6>
                <Button>Save</Button>
            </Column>
        </AccordionItemContent>
    </AccordionItem>
</Accordion>
```

## Keyboard Accessibility

When an accordion is in focus you can toggle it expanded/collapsed with the spacebar or enter keys.
