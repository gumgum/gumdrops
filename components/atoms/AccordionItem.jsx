import React, { Component } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import generateUID from '../utils/generateUID';
import charCodes from '../../constants/charCodes';

class AccordionItem extends Component {
    state = {
        isOpen: this.props.isOpen
    };

    uid = generateUID(this);

    toggleOpen = event => {
        const { type, charCode } = event;
        event.stopPropagation(); // don't want nested accordions to propagate events

        if (this.props.isLocked) return;

        if (
            type === 'keypress' &&
            (charCode === charCodes.SPACEBAR || charCode === charCodes.ENTER)
        ) {
            event.preventDefault(); // prevents page scroll from space key
            this.setState(({ isOpen }) => ({ isOpen: !isOpen }));
        } else if (type === 'click') {
            this.setState(({ isOpen }) => ({ isOpen: !isOpen }));
        }
    };

    render() {
        const { size, context, className, children, label, isLocked } = this.props;

        const { isOpen } = this.state;

        const rootClass = cx('gds-accordion__item', className, {
            [`gds-accordion__item--${context}`]: context,
            'gds-accordion__item--active': isOpen
        });

        const titleClass = cx('gds-accordion__item-title', {
            [`gds-accordion__item-title--${size}`]: size
        });

        const iconClass = cx('gds-accordion__item-icon', {
            [`gds-accordion__item-icon--${size}`]: size
        });

        const childClass = cx('gds-accordion__child-items', {
            [`gds-accordion__child-items--${size}`]: size
        });

        const newChildren = React.Children.map(children, child => {
            return React.cloneElement(child, {
                context,
                size
            });
        });

        const regionId = `AccordionItem-region-${this.uid}`;
        const labelId = `AccordionItem-label-${this.uid}`;

        return (
            <li
                aria-pressed={isOpen}
                aria-controls={regionId}
                aria-expanded={isOpen}
                className={rootClass}
                onClick={this.toggleOpen}
                onKeyPress={this.toggleOpen}
                role="button"
                tabIndex={isLocked ? -1 : 0}>
                <h4 id={labelId} className={titleClass}>
                    {label}
                </h4>
                {!isLocked && <span className={iconClass} />}
                <ul
                    aria-labelledby={labelId}
                    aria-hidden={!isOpen}
                    className={childClass}
                    id={regionId}
                    role="region">
                    {newChildren}
                </ul>
            </li>
        );
    }
}

AccordionItem.displayName = 'AccordionItem';

AccordionItem.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
    context: PropTypes.string,
    isOpen: PropTypes.bool,
    isLocked: PropTypes.bool,
    label: PropTypes.string,
    size: PropTypes.string
};

export default AccordionItem;
