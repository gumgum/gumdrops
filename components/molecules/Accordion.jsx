import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

const Accordion = ({ children, context, size, className, initialAllOpen, ...otherProps }) => {
    const rootClass = cx('gds-accordion', className, {
        [`gds-accordion--${context}`]: context
    });

    const newChildren = React.Children.map(children, child => {
        return React.cloneElement(child, {
            context,
            size,
            isOpen: child.props.isOpen ? child.props.isOpen : initialAllOpen
        });
    });

    return (
        <div className={rootClass} {...otherProps}>
            <ul className="gds-accordion-list">{newChildren}</ul>
        </div>
    );
};

Accordion.displayName = 'Accordion';

Accordion.propTypes = {
    children: PropTypes.node.isRequired,
    /** dark, white */
    context: PropTypes.string,
    initialAllOpen: PropTypes.bool,
    /** sm */
    size: PropTypes.oneOf(['sm']),
    className: PropTypes.string
};

export default Accordion;
