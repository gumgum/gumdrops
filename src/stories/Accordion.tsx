import React from 'react';
import cx from 'classnames';

interface AccordionCommonProps {
    color?: 'dark' | 'white';
    size?: 'sm' | '';
}

type OmitColor<T> = Omit<T, 'color'>;

export interface AccordionItemProps
    extends AccordionCommonProps,
        OmitColor<React.LiHTMLAttributes<HTMLLIElement>> {
    isOpen?: boolean;
    isLocked?: boolean;
    label: React.ReactChild;
}

export interface AccordionProps
    extends AccordionCommonProps,
        OmitColor<React.HTMLAttributes<Element>> {
    initialAllOpen?: boolean;
}

export const Accordion: React.FC<AccordionProps> = ({
    children,
    color,
    size = '',
    className,
    initialAllOpen,
    ...otherProps
}) => {
    const rootClass = cx('gds-accordion', className, {
        [`gds-accordion--${color}`]: color
    });

    const newChildren = React.Children.map(children, child => {
        if (!React.isValidElement<AccordionItemProps>(child)) {
            return child;
        }

        return React.cloneElement<AccordionItemProps>(child, {
            color,
            size,
            isOpen: child.props.isOpen ? child.props.isOpen : initialAllOpen
        });
    });

    return (
        <div className={rootClass} {...otherProps}>
            <ul className="gds-accordion-list">{newChildren}</ul>
        </div>
    );
};
