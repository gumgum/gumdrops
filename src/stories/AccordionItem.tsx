import React, { MouseEvent, KeyboardEvent, useState } from 'react';
import cx from 'classnames';
import { AccordionItemProps } from './Accordion';
import { keyCodes } from 'types';

type ToggleEvent = MouseEvent<HTMLLIElement> | KeyboardEvent<HTMLLIElement>;

export const AccordionItem: React.FC<AccordionItemProps> = ({
    isOpen: initialIsOpen,
    isLocked,
    label,
    size,
    color,
    className,
    children
}) => {
    const [isOpen, setIsOpen] = useState(initialIsOpen);

    const toggleOpen = (event: ToggleEvent): void => {
        event.stopPropagation(); // don't want nested accordions to propagate events
        event.preventDefault(); // prevents page scroll from space key
        const { type } = event;
        const { key } = event as KeyboardEvent;

        if (isLocked) return;
        const allowToggle =
            (type === 'keypress' && [keyCodes.space, keyCodes.enter].includes(key as keyCodes)) ||
            type === 'click';

        if (!allowToggle) return;

        setIsOpen(isOpen => !isOpen);
    };

    const rootClass = cx('gds-accordion__item', className, {
        [`gds-accordion__item--${color}`]: color,
        'gds-accordion__item--active': isOpen
    });

    const titleClass = cx('gds-accordion__item-title', {
        [`gds-accordion__item-title--${size}`]: size
    });

    const iconClass = cx('gds-accordion__item-icon', {
        [`gds-accordion__item-icon--${size}`]: size
    });

    const childClass = cx('gds-accordion__child-items', {
        [`gds-accordion__child-items--${size}`]: size
    });

    const newChildren = React.Children.map(children, child => {
        if (!React.isValidElement<AccordionItemProps>(child)) {
            return child;
        }

        return React.cloneElement<AccordionItemProps>(child, {
            color,
            size
        });
    });

    const regionId = `AccordionItem-region-${label}`;
    const labelId = `AccordionItem-label-${label}`;

    return (
        <li
            aria-pressed={isOpen}
            aria-controls={regionId}
            aria-expanded={isOpen}
            className={rootClass}
            onClick={toggleOpen}
            onKeyPress={toggleOpen}
            role="button"
            tabIndex={isLocked ? -1 : 0}>
            <h4 id={labelId} className={titleClass}>
                {label}
            </h4>
            {!isLocked && <span className={iconClass} />}
            <ul
                aria-labelledby={labelId}
                aria-hidden={!isOpen}
                className={childClass}
                id={regionId}
                role="region">
                {newChildren}
            </ul>
        </li>
    );
};
