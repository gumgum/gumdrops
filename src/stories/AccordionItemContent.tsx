import React from 'react';
import cx from 'classnames';
import { AccordionItemProps } from './Accordion';

export const AccordionItemContent: React.FC<Omit<AccordionItemProps, 'label'>> = ({
    className,
    color,
    size,
    children,
    ...otherProps
}) => {
    const stopClick: React.MouseEventHandler = e => {
        e.stopPropagation();
    };

    const rootClass = cx('gds-accordion__child-item', className, {
        [`gds-accordion__child-item--${color}`]: color
    });

    const titleClass = cx('gds-accordion__child-item-title', {
        [`gds-accordion__child-item-title--${size}`]: size
    });

    return (
        <li className={rootClass} onClick={stopClick} {...otherProps}>
            <h4 className={titleClass}>{children}</h4>
        </li>
    );
};
