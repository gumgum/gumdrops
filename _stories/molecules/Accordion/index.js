import React from 'react';
import { text } from '@storybook/addon-knobs';
import { optionalSelect } from '../../../components/utils/optionalSelect';

import readme from './README.md';

import Accordion from '../../../components/molecules/Accordion';
import AccordionItem from '../../../components/atoms/AccordionItem';
import AccordionItemContent from '../../../components/atoms/AccordionItemContent';

const sizeOptions = {
    sm: 'sm',
    'No Value': ''
};

const contextOptions = {
    dark: 'dark',
    white: 'white',
    'No Value': ''
};

const component = () => (
    <Accordion
        size={optionalSelect('Size', sizeOptions, '')}
        context={optionalSelect('Context', contextOptions, '')}
        className={text('Class', '')}>
        <AccordionItem label="I'm locked open" isLocked isOpen>
            <AccordionItemContent>Locked Content</AccordionItemContent>
        </AccordionItem>
        <AccordionItem label={text('Item Label 1', 'Item 1')}>
            <AccordionItemContent>Content 1</AccordionItemContent>
        </AccordionItem>
        <AccordionItem label={text('Item Label 2', 'Item 2')}>
            <AccordionItemContent>Content 2</AccordionItemContent>
        </AccordionItem>
        <AccordionItem label={text('Item Label 3', 'Item 3')}>
            <AccordionItemContent>Content 3</AccordionItemContent>
            <Accordion>
                <AccordionItem label="Nested Item">
                    <AccordionItemContent>Nested Content</AccordionItemContent>
                </AccordionItem>
            </Accordion>
        </AccordionItem>
    </Accordion>
);

export default [readme, component];
