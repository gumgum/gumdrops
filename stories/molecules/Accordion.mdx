import { Canvas, Source } from '@storybook/blocks';
import { ArgsTable } from '@storybook/addon-docs';

import * as AccordionStories from './Accordion.stories';
import Accordion from '../../components/molecules/Accordion';
import AccordionItem from '../../components/atoms/AccordionItem';
import AccordionItemContent from '../../components/atoms/AccordionItemContent';

# Accordion

The `<Accordion>` component is a collapsible container for holding related elements. Nest `<AccordionItem>` components inside `<Accordion>` for each drawer. Nest `<AccordionItemContent>` in each `<AccordionItem>` if you would like to display listed information within an open `<AccordionItem>`. Each `<Accordion>` related component accepts a className and otherProps so you are able to further customize the component.

## Examples

You can use `<Accordion>` to display any information that you pass in, such as:

```jsx
<Accordion>
    <AccordionItem label="Controls">
        <FormGroup>
            <FormGroupLabel text="Name" />
            <TextInput name="name" />
        </FormGroup>
    </AccordionItem>
</Accordion>
```

You can also use `<Accordion>` to display information in a list form, in the case you want to list items in rows. Example:

```jsx
<Accordion>
    <AccordionItem label="Settings">
        <AccordionItemContent>
            <Column md="6">
                Setting 1
            </Column>
            <Column md=6>
                <Button>Save</Button>
            </Column>
        </AccordionItemContent>
        <AccordionItemContent>
            <Column md="6">
                Setting 2
            </Column>
            <Column md=6>
                <Button>Save</Button>
            </Column>
        </AccordionItemContent>
    </AccordionItem>
</Accordion>
```

## Keyboard Accessibility

When an accordion is in focus, you can toggle it expanded/collapsed with the spacebar or enter keys.

## Component Example

<Canvas of={AccordionStories.Default} />

## Component Example Source

<Source of={AccordionStories.Default} />

## Accordion Props

<ArgsTable of={Accordion} />

## AccordionItem Props

<ArgsTable of={AccordionItem} />

## AccordionItemContent Props

<ArgsTable of={AccordionItemContent} />
