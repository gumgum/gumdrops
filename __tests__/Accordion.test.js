import React from 'react';
import user from '@testing-library/user-event';
import { render, screen } from '@testing-library/react';
import { Accordion } from 'stories/Accordion';
import { AccordionItem } from 'stories/AccordionItem';
import { AccordionItemContent } from 'stories/AccordionItemContent';

describe('Accordion', () => {
    it('renders', async () => {
        const { container } = render(
            <Accordion>
                <AccordionItem label="I'm locked and open" isLocked isOpen>
                    <AccordionItemContent>Locked Content</AccordionItemContent>
                </AccordionItem>
                <AccordionItem label="Item 1">
                    <AccordionItemContent>Content 1</AccordionItemContent>
                </AccordionItem>
                <AccordionItem label="Item 2">
                    <AccordionItemContent>Content 2</AccordionItemContent>
                </AccordionItem>
                <AccordionItem label="Item 3">
                    <AccordionItemContent>Content 3</AccordionItemContent>
                    <Accordion>
                        <AccordionItem label="Nested Item">
                            <AccordionItemContent>Nested Content</AccordionItemContent>
                        </AccordionItem>
                    </Accordion>
                </AccordionItem>
            </Accordion>
        );
        screen.getByRole('list');
        screen.getByText('Locked Content');
        user.click(screen.getByText("I'm locked and open"));
        screen.getByText('Locked Content');
        const item1 = screen.getByRole('button', { name: 'Item 1' });
        const item2 = screen.getByRole('button', { name: 'Item 2' });
        const item3 = screen.getByRole('button', { name: 'Item 3' });
        expect(container.firstChild).toMatchSnapshot();
        // NOTE: these tests are commented out because of this issue:
        // https://github.com/testing-library/dom-testing-library/issues/875
        //expect(within(item1).queryByText('Content 1')).toBe(null);
        //expect(within(item2).queryByText('Content 2')).toBe(null);
        //expect(within(item3).queryByText('Content 3')).toBe(null);
        //expect(within(item3).queryByText('Nested Content')).toBe(null);
        user.click(item1);
        user.click(item2);
        user.click(item3);
        screen.getByText('Content 1');
        screen.getByText('Content 2');
        screen.getByText('Content 3');
        screen.getByText('Nested Content');
        expect(container.firstChild).toMatchSnapshot();
        user.click(item1);
        user.click(item2);
        user.click(item3);
        user.click(container);
        expect(container.firstChild).toMatchSnapshot();
        //expect(within(item1).queryByText('Content 1')).toBe(null);
        //expect(within(item2).queryByText('Content 2')).toBe(null);
        //expect(within(item3).queryByText('Content 3')).toBe(null);
        //expect(within(item3).queryByText('Nested Content')).toBe(null);
        user.tab();
        user.tab();
        expect(item2).toHaveFocus();
        user.type(item2, '{enter}');
        screen.getByText('Content 2');
        user.type(item2, '{space}');
        //expect(within(item2).queryByText('Content 2')).toBe(null);
        expect(container.firstChild).toMatchSnapshot();
    });
});
