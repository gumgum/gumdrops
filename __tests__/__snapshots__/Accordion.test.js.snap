// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion renders 1`] = `
<div
  class="gds-accordion"
>
  <ul
    class="gds-accordion-list"
  >
    <li
      aria-controls="AccordionItem-region-I'm locked and open"
      aria-expanded="true"
      aria-pressed="true"
      class="gds-accordion__item gds-accordion__item--active"
      role="button"
      tabindex="-1"
    >
      <h4
        class="gds-accordion__item-title"
        id="AccordionItem-label-I'm locked and open"
      >
        I'm locked and open
      </h4>
      <ul
        aria-hidden="false"
        aria-labelledby="AccordionItem-label-I'm locked and open"
        class="gds-accordion__child-items"
        id="AccordionItem-region-I'm locked and open"
        role="region"
      >
        <li
          class="gds-accordion__child-item"
        >
          <h4
            class="gds-accordion__child-item-title"
          >
            Locked Content
          </h4>
        </li>
      </ul>
    </li>
    <li
      aria-controls="AccordionItem-region-Item 1"
      class="gds-accordion__item"
      role="button"
      tabindex="0"
    >
      <h4
        class="gds-accordion__item-title"
        id="AccordionItem-label-Item 1"
      >
        Item 1
      </h4>
      <span
        class="gds-accordion__item-icon"
      />
      <ul
        aria-hidden="true"
        aria-labelledby="AccordionItem-label-Item 1"
        class="gds-accordion__child-items"
        id="AccordionItem-region-Item 1"
        role="region"
      >
        <li
          class="gds-accordion__child-item"
        >
          <h4
            class="gds-accordion__child-item-title"
          >
            Content 1
          </h4>
        </li>
      </ul>
    </li>
    <li
      aria-controls="AccordionItem-region-Item 2"
      class="gds-accordion__item"
      role="button"
      tabindex="0"
    >
      <h4
        class="gds-accordion__item-title"
        id="AccordionItem-label-Item 2"
      >
        Item 2
      </h4>
      <span
        class="gds-accordion__item-icon"
      />
      <ul
        aria-hidden="true"
        aria-labelledby="AccordionItem-label-Item 2"
        class="gds-accordion__child-items"
        id="AccordionItem-region-Item 2"
        role="region"
      >
        <li
          class="gds-accordion__child-item"
        >
          <h4
            class="gds-accordion__child-item-title"
          >
            Content 2
          </h4>
        </li>
      </ul>
    </li>
    <li
      aria-controls="AccordionItem-region-Item 3"
      class="gds-accordion__item"
      role="button"
      tabindex="0"
    >
      <h4
        class="gds-accordion__item-title"
        id="AccordionItem-label-Item 3"
      >
        Item 3
      </h4>
      <span
        class="gds-accordion__item-icon"
      />
      <ul
        aria-hidden="true"
        aria-labelledby="AccordionItem-label-Item 3"
        class="gds-accordion__child-items"
        id="AccordionItem-region-Item 3"
        role="region"
      >
        <li
          class="gds-accordion__child-item"
        >
          <h4
            class="gds-accordion__child-item-title"
          >
            Content 3
          </h4>
        </li>
        <div
          class="gds-accordion"
        >
          <ul
            class="gds-accordion-list"
          >
            <li
              aria-controls="AccordionItem-region-Nested Item"
              class="gds-accordion__item"
              role="button"
              tabindex="0"
            >
              <h4
                class="gds-accordion__item-title"
                id="AccordionItem-label-Nested Item"
              >
                Nested Item
              </h4>
              <span
                class="gds-accordion__item-icon"
              />
              <ul
                aria-hidden="true"
                aria-labelledby="AccordionItem-label-Nested Item"
                class="gds-accordion__child-items"
                id="AccordionItem-region-Nested Item"
                role="region"
              >
                <li
                  class="gds-accordion__child-item"
                >
                  <h4
                    class="gds-accordion__child-item-title"
                  >
                    Nested Content
                  </h4>
                </li>
              </ul>
            </li>
          </ul>
        </div>
      </ul>
    </li>
  </ul>
</div>
`;

exports[`Accordion renders 2`] = `
<div
  class="gds-accordion"
>
  <ul
    class="gds-accordion-list"
  >
    <li
      aria-controls="AccordionItem-region-I'm locked and open"
      aria-expanded="true"
      aria-pressed="true"
      class="gds-accordion__item gds-accordion__item--active"
      role="button"
      tabindex="-1"
    >
      <h4
        class="gds-accordion__item-title"
        id="AccordionItem-label-I'm locked and open"
      >
        I'm locked and open
      </h4>
      <ul
        aria-hidden="false"
        aria-labelledby="AccordionItem-label-I'm locked and open"
        class="gds-accordion__child-items"
        id="AccordionItem-region-I'm locked and open"
        role="region"
      >
        <li
          class="gds-accordion__child-item"
        >
          <h4
            class="gds-accordion__child-item-title"
          >
            Locked Content
          </h4>
        </li>
      </ul>
    </li>
    <li
      aria-controls="AccordionItem-region-Item 1"
      aria-expanded="true"
      aria-pressed="true"
      class="gds-accordion__item gds-accordion__item--active"
      role="button"
      tabindex="0"
    >
      <h4
        class="gds-accordion__item-title"
        id="AccordionItem-label-Item 1"
      >
        Item 1
      </h4>
      <span
        class="gds-accordion__item-icon"
      />
      <ul
        aria-hidden="false"
        aria-labelledby="AccordionItem-label-Item 1"
        class="gds-accordion__child-items"
        id="AccordionItem-region-Item 1"
        role="region"
      >
        <li
          class="gds-accordion__child-item"
        >
          <h4
            class="gds-accordion__child-item-title"
          >
            Content 1
          </h4>
        </li>
      </ul>
    </li>
    <li
      aria-controls="AccordionItem-region-Item 2"
      aria-expanded="true"
      aria-pressed="true"
      class="gds-accordion__item gds-accordion__item--active"
      role="button"
      tabindex="0"
    >
      <h4
        class="gds-accordion__item-title"
        id="AccordionItem-label-Item 2"
      >
        Item 2
      </h4>
      <span
        class="gds-accordion__item-icon"
      />
      <ul
        aria-hidden="false"
        aria-labelledby="AccordionItem-label-Item 2"
        class="gds-accordion__child-items"
        id="AccordionItem-region-Item 2"
        role="region"
      >
        <li
          class="gds-accordion__child-item"
        >
          <h4
            class="gds-accordion__child-item-title"
          >
            Content 2
          </h4>
        </li>
      </ul>
    </li>
    <li
      aria-controls="AccordionItem-region-Item 3"
      aria-expanded="true"
      aria-pressed="true"
      class="gds-accordion__item gds-accordion__item--active"
      role="button"
      tabindex="0"
    >
      <h4
        class="gds-accordion__item-title"
        id="AccordionItem-label-Item 3"
      >
        Item 3
      </h4>
      <span
        class="gds-accordion__item-icon"
      />
      <ul
        aria-hidden="false"
        aria-labelledby="AccordionItem-label-Item 3"
        class="gds-accordion__child-items"
        id="AccordionItem-region-Item 3"
        role="region"
      >
        <li
          class="gds-accordion__child-item"
        >
          <h4
            class="gds-accordion__child-item-title"
          >
            Content 3
          </h4>
        </li>
        <div
          class="gds-accordion"
        >
          <ul
            class="gds-accordion-list"
          >
            <li
              aria-controls="AccordionItem-region-Nested Item"
              class="gds-accordion__item"
              role="button"
              tabindex="0"
            >
              <h4
                class="gds-accordion__item-title"
                id="AccordionItem-label-Nested Item"
              >
                Nested Item
              </h4>
              <span
                class="gds-accordion__item-icon"
              />
              <ul
                aria-hidden="true"
                aria-labelledby="AccordionItem-label-Nested Item"
                class="gds-accordion__child-items"
                id="AccordionItem-region-Nested Item"
                role="region"
              >
                <li
                  class="gds-accordion__child-item"
                >
                  <h4
                    class="gds-accordion__child-item-title"
                  >
                    Nested Content
                  </h4>
                </li>
              </ul>
            </li>
          </ul>
        </div>
      </ul>
    </li>
  </ul>
</div>
`;

exports[`Accordion renders 3`] = `
<div
  class="gds-accordion"
>
  <ul
    class="gds-accordion-list"
  >
    <li
      aria-controls="AccordionItem-region-I'm locked and open"
      aria-expanded="true"
      aria-pressed="true"
      class="gds-accordion__item gds-accordion__item--active"
      role="button"
      tabindex="-1"
    >
      <h4
        class="gds-accordion__item-title"
        id="AccordionItem-label-I'm locked and open"
      >
        I'm locked and open
      </h4>
      <ul
        aria-hidden="false"
        aria-labelledby="AccordionItem-label-I'm locked and open"
        class="gds-accordion__child-items"
        id="AccordionItem-region-I'm locked and open"
        role="region"
      >
        <li
          class="gds-accordion__child-item"
        >
          <h4
            class="gds-accordion__child-item-title"
          >
            Locked Content
          </h4>
        </li>
      </ul>
    </li>
    <li
      aria-controls="AccordionItem-region-Item 1"
      aria-expanded="false"
      aria-pressed="false"
      class="gds-accordion__item"
      role="button"
      tabindex="0"
    >
      <h4
        class="gds-accordion__item-title"
        id="AccordionItem-label-Item 1"
      >
        Item 1
      </h4>
      <span
        class="gds-accordion__item-icon"
      />
      <ul
        aria-hidden="true"
        aria-labelledby="AccordionItem-label-Item 1"
        class="gds-accordion__child-items"
        id="AccordionItem-region-Item 1"
        role="region"
      >
        <li
          class="gds-accordion__child-item"
        >
          <h4
            class="gds-accordion__child-item-title"
          >
            Content 1
          </h4>
        </li>
      </ul>
    </li>
    <li
      aria-controls="AccordionItem-region-Item 2"
      aria-expanded="false"
      aria-pressed="false"
      class="gds-accordion__item"
      role="button"
      tabindex="0"
    >
      <h4
        class="gds-accordion__item-title"
        id="AccordionItem-label-Item 2"
      >
        Item 2
      </h4>
      <span
        class="gds-accordion__item-icon"
      />
      <ul
        aria-hidden="true"
        aria-labelledby="AccordionItem-label-Item 2"
        class="gds-accordion__child-items"
        id="AccordionItem-region-Item 2"
        role="region"
      >
        <li
          class="gds-accordion__child-item"
        >
          <h4
            class="gds-accordion__child-item-title"
          >
            Content 2
          </h4>
        </li>
      </ul>
    </li>
    <li
      aria-controls="AccordionItem-region-Item 3"
      aria-expanded="false"
      aria-pressed="false"
      class="gds-accordion__item"
      role="button"
      tabindex="0"
    >
      <h4
        class="gds-accordion__item-title"
        id="AccordionItem-label-Item 3"
      >
        Item 3
      </h4>
      <span
        class="gds-accordion__item-icon"
      />
      <ul
        aria-hidden="true"
        aria-labelledby="AccordionItem-label-Item 3"
        class="gds-accordion__child-items"
        id="AccordionItem-region-Item 3"
        role="region"
      >
        <li
          class="gds-accordion__child-item"
        >
          <h4
            class="gds-accordion__child-item-title"
          >
            Content 3
          </h4>
        </li>
        <div
          class="gds-accordion"
        >
          <ul
            class="gds-accordion-list"
          >
            <li
              aria-controls="AccordionItem-region-Nested Item"
              class="gds-accordion__item"
              role="button"
              tabindex="0"
            >
              <h4
                class="gds-accordion__item-title"
                id="AccordionItem-label-Nested Item"
              >
                Nested Item
              </h4>
              <span
                class="gds-accordion__item-icon"
              />
              <ul
                aria-hidden="true"
                aria-labelledby="AccordionItem-label-Nested Item"
                class="gds-accordion__child-items"
                id="AccordionItem-region-Nested Item"
                role="region"
              >
                <li
                  class="gds-accordion__child-item"
                >
                  <h4
                    class="gds-accordion__child-item-title"
                  >
                    Nested Content
                  </h4>
                </li>
              </ul>
            </li>
          </ul>
        </div>
      </ul>
    </li>
  </ul>
</div>
`;

exports[`Accordion renders 4`] = `
<div
  class="gds-accordion"
>
  <ul
    class="gds-accordion-list"
  >
    <li
      aria-controls="AccordionItem-region-I'm locked and open"
      aria-expanded="true"
      aria-pressed="true"
      class="gds-accordion__item gds-accordion__item--active"
      role="button"
      tabindex="-1"
    >
      <h4
        class="gds-accordion__item-title"
        id="AccordionItem-label-I'm locked and open"
      >
        I'm locked and open
      </h4>
      <ul
        aria-hidden="false"
        aria-labelledby="AccordionItem-label-I'm locked and open"
        class="gds-accordion__child-items"
        id="AccordionItem-region-I'm locked and open"
        role="region"
      >
        <li
          class="gds-accordion__child-item"
        >
          <h4
            class="gds-accordion__child-item-title"
          >
            Locked Content
          </h4>
        </li>
      </ul>
    </li>
    <li
      aria-controls="AccordionItem-region-Item 1"
      aria-expanded="false"
      aria-pressed="false"
      class="gds-accordion__item"
      role="button"
      tabindex="0"
    >
      <h4
        class="gds-accordion__item-title"
        id="AccordionItem-label-Item 1"
      >
        Item 1
      </h4>
      <span
        class="gds-accordion__item-icon"
      />
      <ul
        aria-hidden="true"
        aria-labelledby="AccordionItem-label-Item 1"
        class="gds-accordion__child-items"
        id="AccordionItem-region-Item 1"
        role="region"
      >
        <li
          class="gds-accordion__child-item"
        >
          <h4
            class="gds-accordion__child-item-title"
          >
            Content 1
          </h4>
        </li>
      </ul>
    </li>
    <li
      aria-controls="AccordionItem-region-Item 2"
      aria-expanded="true"
      aria-pressed="true"
      class="gds-accordion__item gds-accordion__item--active"
      role="button"
      tabindex="0"
    >
      <h4
        class="gds-accordion__item-title"
        id="AccordionItem-label-Item 2"
      >
        Item 2
      </h4>
      <span
        class="gds-accordion__item-icon"
      />
      <ul
        aria-hidden="false"
        aria-labelledby="AccordionItem-label-Item 2"
        class="gds-accordion__child-items"
        id="AccordionItem-region-Item 2"
        role="region"
      >
        <li
          class="gds-accordion__child-item"
        >
          <h4
            class="gds-accordion__child-item-title"
          >
            Content 2
          </h4>
        </li>
      </ul>
    </li>
    <li
      aria-controls="AccordionItem-region-Item 3"
      aria-expanded="false"
      aria-pressed="false"
      class="gds-accordion__item"
      role="button"
      tabindex="0"
    >
      <h4
        class="gds-accordion__item-title"
        id="AccordionItem-label-Item 3"
      >
        Item 3
      </h4>
      <span
        class="gds-accordion__item-icon"
      />
      <ul
        aria-hidden="true"
        aria-labelledby="AccordionItem-label-Item 3"
        class="gds-accordion__child-items"
        id="AccordionItem-region-Item 3"
        role="region"
      >
        <li
          class="gds-accordion__child-item"
        >
          <h4
            class="gds-accordion__child-item-title"
          >
            Content 3
          </h4>
        </li>
        <div
          class="gds-accordion"
        >
          <ul
            class="gds-accordion-list"
          >
            <li
              aria-controls="AccordionItem-region-Nested Item"
              class="gds-accordion__item"
              role="button"
              tabindex="0"
            >
              <h4
                class="gds-accordion__item-title"
                id="AccordionItem-label-Nested Item"
              >
                Nested Item
              </h4>
              <span
                class="gds-accordion__item-icon"
              />
              <ul
                aria-hidden="true"
                aria-labelledby="AccordionItem-label-Nested Item"
                class="gds-accordion__child-items"
                id="AccordionItem-region-Nested Item"
                role="region"
              >
                <li
                  class="gds-accordion__child-item"
                >
                  <h4
                    class="gds-accordion__child-item-title"
                  >
                    Nested Content
                  </h4>
                </li>
              </ul>
            </li>
          </ul>
        </div>
      </ul>
    </li>
  </ul>
</div>
`;
